lib/Sema/SemaExpr.cpp

```diff
@@ -8058,6 +8058,112 @@ Sema::CheckSingleAssignmentConstraints(QualType LHSType, ExprResult &CallerRHS,
   return result;
 }
 
+/// If a declaration has a Checked C bounds-safe interface
+/// attached to it, return the checked pointer type for the interface.
+/// Otherwise return a null QualType.
+///
+/// This falls into two cases:
```

Comment:
Feel free to tell me otherwise, but I'm wondering if this logic belongs here, or if it belongs in helpers on the declaration objects themselves?  Perhaps I need to get used to the world of compilers--I'm used to going with stronger encapsulation/containment.